scripts/ci/ci_pylint.sh

```diff
@@ -17,14 +17,53 @@
 # limitations under the License.
 #
 
-# Script to run Pylint on all code. Should be started from root directory of Airflow:
+# Script to run Pylint on all code. Can be started from any working directory
 # ./scripts/ci/ci_pylint.sh
 
-set -ex
+set -uo pipefail
+
+# Uncomment to see the commands executed
+# set -x
+
+MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
I thought `BASH_SOURCE` was only for when you did `source myscript` not `./myscript` - am I mis-remembering?

(BTW `git` has `git rev-parse --show-toplevel` which will print the abs-path to the root of the repo.)

Reply:
Not only - BASH_SOURCE array contains all bash scripts that were used. I always use this construct in all the bash scripts I run - this way I get them independent from where they are run.

See here for example: https://stackoverflow.com/questions/59895/get-the-source-directory-of-a-bash-script-from-within-the-script-itself

The problem with git is that not everywhere you have .git directory. For example we've learned that Dockerhub does not have the whole .git repo while building Docker. It's very similar in case of cloud build - there is no guarantee you have .git repo around. Of course in our case we know we have the repo in Travis, but not relying on it if we can is a better strategy IMHO.

Reply:
Just to backup my statement on cloud build: https://github.com/GoogleCloudPlatform/cloud-builders/issues/236